### PR TITLE
Form validation on course number

### PR DESCRIPTION
**Pull request by asadazam93**
### Description
Added form validation on course number. Course number should not consist of any special characters or spaces except periods and underscores. This validation is used in studio course number as well so publisher needs to implement the same validation to keep the data consistent.

**Sandbox**
- N/A

### Reviewers
If you've been tagged for review, please check your corresponding box once you've given the :+1:.
- [ ] @Rabia23   
- [ ] @attiyaIshaque  

### Post-review
- [ ] Rebase and squash commits

### CODE REVIEW

File: course_discovery/conf/locale/en/LC_MESSAGES/django.po [hunk 1]
```diff
@@ -7,7 +7,7 @@ msgid ""
 msgstr ""
 "Project-Id-Version: PACKAGE VERSION\n"
 "Report-Msgid-Bugs-To: \n"
-"POT-Creation-Date: 2017-10-16 17:29+0000\n"
+"POT-Creation-Date: 2017-10-25 06:07+0000\n"
```
Comment (attiyaIshaque) [inline course_discovery/conf/locale/en/LC_MESSAGES/django.po:10]: Why you change the date?

File: course_discovery/apps/publisher/forms.py [hunk 2]
```diff
@@ -191,6 +192,15 @@ def clean_title(self):
         """
         return html.unescape(self.cleaned_data.get("title"))
 
+    def clean_number(self):
+        """
+        Validate that number doesnt consist of any special characters
```
Comment (attiyaIshaque) [inline course_discovery/apps/publisher/forms.py]: Typo, should be s/doesnt/doesn't.

File: course_discovery/apps/publisher/forms.py [hunk 3]
```diff
@@ -1,7 +1,17 @@
...
+import re
 
 import waffle
+from course_discovery.apps.course_metadata.choices import CourseRunPacing
+from course_discovery.apps.course_metadata.models import LevelType, Organization, Person, Subject
+from course_discovery.apps.ietf_language_tags.models import LanguageTag
+from course_discovery.apps.publisher.mixins import LanguageModelSelect2Multiple, get_user_organizations
+from course_discovery.apps.publisher.models import (
+    Course, CourseRun, CourseUserRole, OrganizationExtension, OrganizationUserRole, PublisherUser, Seat, User
+)
```
Comment (attiyaIshaque) [inline course_discovery/apps/publisher/forms.py]: I think Imports are incorrectly sorted, please sort it again. :)

**Review by attiyaIshaque — CHANGES_REQUESTED** [review]

File: course_discovery/conf/locale/en/LC_MESSAGES/django.po (hunk 1 above)
Comment (asadazam93) [reply course_discovery/conf/locale/en/LC_MESSAGES/django.po:10]: I updated the translation files so the date has been updated automatically.

File: course_discovery/apps/publisher/forms.py (hunk 3 above)
Comment (asadazam93) [reply course_discovery/apps/publisher/forms.py]: I used isort to sort the imports still they are incorrectly sorted, I don't know why

File: course_discovery/conf/locale/en/LC_MESSAGES/django.po (hunk 1 above)
Comment (attiyaIshaque) [reply course_discovery/conf/locale/en/LC_MESSAGES/django.po:10]: Okay :)

File: course_discovery/apps/publisher/forms.py (hunk 3 above)
Comment (attiyaIshaque) [reply course_discovery/apps/publisher/forms.py]: you can sort it on your own way manually.

**Review by attiyaIshaque — APPROVED** [review]
LGTM 👍

File: course_discovery/apps/publisher/tests/test_forms.py [hunk 4]
```diff
@@ -246,6 +246,22 @@ def test_duplicate_number(self):
...
 
+    def test_invalid_number(self):
+        """
+        Verify that clean_number raises 'ValidationError' if the course number consists of special characters
+        or spaces
+        """
+        course_form = CourseForm()
+        course_form.cleaned_data = {'number': '123 a'}
+        with self.assertRaises(ValidationError):
+            course_form.clean_number()
```
Comment (Rabia23) [inline course_discovery/apps/publisher/tests/test_forms.py:257]: As there are many `ValidationError` in the form `clean` method, We should also test if the expected `ValidationError` has occurred.

Comment (asadazam93) [reply course_discovery/apps/publisher/tests/test_forms.py:257]: I am calling the clean_number method which has only one ValidationError

Comment (Rabia23) [reply course_discovery/apps/publisher/tests/test_forms.py:257]: okay 👍

**Review by Rabia23 — APPROVED** [review]

**Pull request merged**